packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -1,18 +1,38 @@
 import { TransactionBaseService } from "./transaction-base-service"
 import {
+  Address,
   Cart,
   Customer,
   Payment,
   PaymentSession,
   PaymentSessionStatus,
+  ShippingMethod,
 } from "../models"
 import { PaymentService } from "medusa-interfaces"
-import { PaymentProviderDataInput } from "../types/payment-collection"
 
 export type Data = Record<string, unknown>
 export type PaymentData = Data
 export type PaymentSessionData = Data
 
+export type PaymentContext = {
```

Comment:
suggestion: to provide more context to the naming, since we're defining a pattern to how plugins should be created.

`${resource}${action}${Request | Response}`

Reply:
This object is a context more than a request object. The consumer can use a part of fully the context or not at all eventually.